src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,220 @@
+                        System.out.println("Are you sure you want to delete this task (Y/n)?");
+                        System.out.println(tasks.get(i));
+
+                        while (true) {
+                            String inputDelete = scanner.nextLine();
+
+                            if (inputDelete.equalsIgnoreCase("y") || inputDelete.isEmpty()) {
+                                tasks.remove(i);
+                                System.out.println("Task has been deleted");
+                                deleted = true;
+                                break;
+                            } else if (inputDelete.equalsIgnoreCase("n")) {
+                                System.out.println("Nothing has changed");
+                                break;
+                            } else {
+                                System.out.println("Invalid input. Please type 'Y' for yes or 'n' for no.");
+                            }
+                        }
+                        break;
+                    }
+                }
+
+                if (!taskFound) {
+                    System.out.println("Task with ID '" + inputId + "' not found. Use list to view all your tasks.");
+                }
+
+                if (deleted || taskFound) {
+                    break;
+                }
+
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid input. Please enter a valid numerical ID.");
+            }
+        }
+    }
+
+    private static void listTasks() {
+        if (tasks.isEmpty()) {
+            System.out.println("Your list is empty. Add new task with 'add' command");
+        } else {
+            for (Task task : tasks) {
+                System.out.println(task.toString());
+            }
+        }
+    }
+
+    private static void viewTask() {
+        boolean exists = false;
+        System.out.println("Type ID of the task");
+        int input = scanner.nextInt();
+        for (Task task : tasks) {
+            if (task.getId() == input) {
+                System.out.println("Task info:");
+                System.out.println(task);
+                exists = true;
+                break;
+            }
+        }
+
+        if (!exists) {
+            System.out.println("Task with ID '" + input + "' not found. Use list to view all your tasks");
+        }
+
+        scanner.nextLine();
+    }
+
+    private static void saveTasks() {
+        Path path = Paths.get(SAVE_FILE);
+        List<String> lines = tasks.stream()
+                .map(task -> task.getName() + "," + task.getPriority() + "," + task.getDueDate())
+                .collect(Collectors.toList());
+        try {
+            Files.write(path, lines, StandardCharsets.UTF_8);
+            System.out.println("List has been saved to " + SAVE_FILE);
+            scanner.close();
+        } catch (IOException e) {
+            System.out.println("Error has occurred while saving tasks: " + e.getMessage());
```

Comment:
IMPROVE: doporucuji podrobnejsi logovani vyjimek i ze stack tracem a v tomto pripade i ukonceni aplikace s chybovym kodem.

Chybove kody v jave: https://stackoverflow.com/questions/2434592/difference-in-system-exit0-system-exit-1-system-exit1-in-java

```suggestion
            System.err.println("Error has occurred while saving tasks: " + e.getMessage());
             e.printStackTrace(System.err);
             System.exit(1);
```